classes/local/util/mustache.php

```diff
@@ -54,12 +54,16 @@ public function __construct(array $options = []) {
         Mustache_Autoloader::register();
 
         $default = [
-            'loader' => new Mustache_Loader_FilesystemLoader($CFG->dirroot.'/'.$CFG->admin.'/tool/pluginskel/skel'),
             'cache' => make_localcache_directory('tool_pluginskel/mustache/'),
             'pragmas' => [Mustache_Engine::PRAGMA_BLOCKS, Mustache_Engine::PRAGMA_ANCHORED_DOT],
             'escape' => 'addslashes',
         ];
 
+        if(empty($options['loader'])) {
```

Comment:
Is this needed? If the loader is passed via options, it should replace the default value via the `array_replace_recursive()`.

Reply:
Yes, it is needed because when $options['loader'] is set but NULL, array_replace_recursive() will replace the default with NULL.

Reply:
This one is a bit exceptional, because in $defaults you had:
'loader' => new Mustache_Loader_FilesystemLoader($CFG->dirroot.'/'.$CFG->admin.'/tool/pluginskel/skel'),

and you can not execute new Mustache_Loader_FilesystemLoader() with non-existing path - it will throw an exception.

I agree with you that this can surely be done in a more elegant way, e.g. for all the options. But I don't see a need for it - e.g. there is no reason to implement it for anything else than loader.

At the end of the day it's your call of course.

Reply:
Correct. And why should not it throw exception if it is given invalid data? That's why I was wondering how could an NULL loader be passed.

Reply:
I'm not saying it should not throw it - we just should not use it with invalid path, which would be the case here, if the location of the templates is different than:
$CFG->dirroot.'/'.$CFG->admin.'/tool/pluginskel/skel'

I think it will be NULL normally (for you, in a plugin), that's how you instantiate it:
public static function instance($logger, $mustachedir = null)
....
$manager->init_templating_engine($mustachedir);

and when it's null, then the default is taken. This is all to keep your existing logic.